packages/fxa-settings/src/components/CardHeader/index.tsx

```diff
@@ -4,41 +4,88 @@
 
 import React from 'react';
 import { FtlMsg } from 'fxa-react/lib/utils';
+import { MozServices } from '../../lib/types';
 
-// TODO improve typing to allow either default or custom or both (but not none)
-export type CardHeaderProps = {
-  headingWithDefaultServiceFtlId: string;
-  headingWithCustomServiceFtlId?: string;
+interface CardHeaderRequiredProps {
   headingText: string;
-  serviceName?: string;
-};
+}
+
+interface CardHeaderDefaultServiceProps extends CardHeaderRequiredProps {
+  headingWithDefaultServiceFtlId: string;
+}
+
+interface CardHeaderCustomServiceProps extends CardHeaderRequiredProps {
+  headingWithCustomServiceFtlId: string;
+  serviceName: Exclude<MozServices, 'Default'>;
+}
+
+interface CardHeaderBasicProps extends CardHeaderRequiredProps {
+  headingTextFtlId: string;
+}
+
+export type CardHeaderProps =
+  | CardHeaderDefaultServiceProps
+  | CardHeaderCustomServiceProps
+  | CardHeaderBasicProps;
+
+function isCustomService(
+  props: CardHeaderProps
+): props is CardHeaderCustomServiceProps {
+  return (
+    (props as CardHeaderCustomServiceProps).headingWithCustomServiceFtlId !==
+    undefined
+  );
+}
+
+function isDefaultService(
+  props: CardHeaderProps
+): props is CardHeaderDefaultServiceProps {
+  const serviceName = (props as CardHeaderCustomServiceProps).serviceName;
+  return (
+    ((props as CardHeaderDefaultServiceProps).headingWithDefaultServiceFtlId !==
+      undefined &&
+      serviceName === undefined) ||
+    serviceName === MozServices.Default
+  );
+}
+
+const CardHeader = (
+  props:
+    | CardHeaderDefaultServiceProps
```

Comment:
Can we say `props: CardHeaderProps` here? or does that seem less clear?

Reply:
Oh nice catch, I'd forgotten to update it. 👍